packages/ipfs-core/src/components/storage.js

```diff
@@ -0,0 +1,301 @@
+'use strict'
+
+const log = require('debug')('ipfs:components:peer:storage')
+const createRepo = require('../runtime/repo-nodejs')
+const getDefaultConfig = require('../runtime/config-nodejs')
+const { ERR_REPO_NOT_INITIALIZED } = require('ipfs-repo').errors
+const uint8ArrayFromString = require('uint8arrays/from-string')
+const uint8ArrayToString = require('uint8arrays/to-string')
+const PeerId = require('peer-id')
+const { mergeOptions } = require('../utils')
+const configService = require('./config')
+const { NotEnabledError } = require('../errors')
+const createLibP2P = require('./libp2p')
+
+class Storage {
+  /**
+   * @private
+   * @param {PeerId} peerId
+   * @param {Keychain} keychain
+   * @param {Repo} repo
+   * @param {Print} print
+   * @param {boolean} isNew
+   */
+  constructor (peerId, keychain, repo, print, isNew) {
+    this.print = print
+    this.peerId = peerId
+    this.keychain = keychain
+    this.repo = repo
+    this.print = print
+    this.isNew = isNew
+  }
+
+  /**
+   *
+   * @param {Options} options
+   */
+  static async start (options) {
+    const { repoAutoMigrate: autoMigrate, repo: inputRepo, print, silent } = options
+
+    const repo = (typeof inputRepo === 'string' || inputRepo == null)
+      ? createRepo({ path: inputRepo, autoMigrate, silent })
+      : inputRepo
+
+    const { peerId, keychain, isNew } = await loadRepo(repo, options)
+
+    return new Storage(peerId, keychain, repo, print, isNew)
+  }
+}
+module.exports = Storage
+
+/**
+ *
+ * @param {Repo} repo
+ * @param {RepoOptions & InitOptions} options
+ * @returns {Promise<{peerId: PeerId, keychain:Keychain, isNew:boolean }>}
+ */
+const loadRepo = async (repo, options) => {
+  const openError = await openRepo(repo)
+  if (openError == null) {
+    // If opened succefully configure repo
+    return { ...await configureRepo(repo, options), isNew: false }
+  } else if (openError.code === ERR_REPO_NOT_INITIALIZED) {
+    if (options.allowNew === false) {
+      throw new NotEnabledError('new repo initialization is not enabled')
```

Comment:
A more descriptive message would help: 'Initialization of new repos disabled by config, pass `blah blah` to create a new repo' or something.

Reply:
This came from 
https://github.com/ipfs/js-ipfs/blob/583503bd35b23bc02f2eca2413a18d33cddccdcd/packages/ipfs-core/src/components/init.js#L86

how about ?

```suggestion
      throw new NotEnabledError('Initialization of new repos disabled by config, pass `config.init.isNew: true` to enable it')
```